db/db_iter.cc

```diff
@@ -877,9 +877,14 @@ bool DBIter::FindValueForCurrentKey() {
     if (timestamp_lb_ != nullptr) {
       // Only needed when timestamp_lb_ is not null
       [[maybe_unused]] const bool ret = ParseKey(&ikey_);
-      saved_ikey_.assign(iter_.key().data(), iter_.key().size());
```

Comment:
This PR eliminates all uses of `saved_ikey_`, correct? If yes, we can remove the member variable itself as well

Reply:
That's a good catch. I removed the `saved_ikey_`.